### DISCUSSION TITLE
What files are supposed to be where per OS?

### DISCUSSION THREAD

**Discussion by johnmccrae** [openssl/openssl#24312] [Q&A] [answered]
I am compiling out v 3.0.9 and am being partially successful. I have compilation working correctly for Windows to include FIPS support but my installer isn't correct, it's missing files (like fips.dll and openssl.cfg). I am working on that part but that brings up the question - how do I know i have all the files I am supposed to have for various OS's? I need to compile this out for all the things. Is there a listing somewhere of what openssl expects to be in what directory per OS? or  can I interrogate a cnf or someething?

**Comment by nhorman** [comment, marked as answer]
The files produced are going to be dependent on what your configuration is at build time, and what your platform is:
libcrypto.[a|lib]/libssl.[a|lib] - should always be produced, and are the static libraries
libcrypto[.so|.dll]/libssl[so|dll] - are the dynamic libs and will be produced if no-shared is not selected during configure
providers/legacy.[so|.dll] - the legacy provider, should be produced if no-legacy is not selected at build time
providers/fips.[so|dll] - the fips provider, produced if enable-fips is provided on the configuration
apps/openssl[.exe] - should always be produced.

There is no configuration file produced, thats up to you to create, based on the examples throughout the code, based on your needs

As for what to include in an installer, thats largely up to you.  If you want to look at an example, there is a pending PR here:
https://github.com/openssl/installer/pull/4
That provides an example

**Comment by johnmccrae** [comment]
Brilliant, thanks very much!

Cheers,
John

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by mattcaswell** [comment]
> There is no configuration file produced, thats up to you to create, based on the examples throughout the code, based on your needs

That's not correct. A config file *is* produced. For most platforms its this one:

https://github.com/openssl/openssl/blob/master/apps/openssl.cnf
